app/src/main/kotlin/compiler/intermediate/ControlFlow.kt

```diff
@@ -65,7 +81,7 @@ object ControlFlow {
                 functionDetailsGenerators[function]!!.genPrologue(),
                 functionDetailsGenerators[function]!!.genEpilogue()
             )
-        }
+        } // TODO: add initialization and finalization of generators
```

Comment:
Am I right, that this comment is obsolete, since generators are initialized and finalized in foreach loops?

Reply:
We still need to add somewhere the code responsible for initialization, resume, and finalization of generators, I mean the code returned by  genInit, genResume and genFinalize in GeneratorDetailsGenerator

Reply:
Got it, so we should probably create separate maps for `initialization` and `finalization` and pass them to linearization phase. We can left the integration for another task if you feel this is out of scope for this task.

Reply:
Yeah, I think we can do this in the next week